test/system/280-update.bats

```diff
@@ -161,4 +161,141 @@ device-write-iops   = /dev/zero:4000 | - | -
+    tests="
+    health-cmd=echo $msg                        | {{.Config.Healthcheck.Test}}             | [CMD-SHELL echo $msg]
+    health-interval=10s                         | {{.Config.Healthcheck.Interval}}         | 10s
+    health-log-destination=$TMP_DIR_HEALTHCHECK | {{.Config.HealthLogDestination}}         | $TMP_DIR_HEALTHCHECK
+    health-max-log-count=20                     | {{.Config.HealthMaxLogCount}}            | 20
+    health-max-log-size=10                      | {{.Config.HealthMaxLogSize}}             | 10
+    health-on-failure=restart                   | {{.Config.HealthcheckOnFailureAction}}   | restart
+    health-retries=5                            | {{.Config.Healthcheck.Retries}}          | 5
+    health-timeout=10s                          | {{.Config.Healthcheck.Timeout}}          | 10s
+    health-start-period=10s                     | {{.Config.Healthcheck.StartPeriod}}      | 10s
+    health-startup-cmd=echo $msg                | {{.Config.StartupHealthCheck.Test}}      | [CMD-SHELL echo $msg]
+    health-startup-interval=10s                 | {{.Config.StartupHealthCheck.Interval}}  | 10s
+    health-startup-retries=5                    | {{.Config.StartupHealthCheck.Retries}}   | 5
+    health-startup-success=10                   | {{.Config.StartupHealthCheck.Successes}} | 10
+    health-startup-timeout=10s                  | {{.Config.StartupHealthCheck.Timeout}}   | 10s
+    "
+
+    run_podman run -d --name $ctrname $IMAGE top
+    cid="$output"
+
+    # Pass 1: read the table above, gather up the options, format and expected values
+    local -a opts
+    local -A formats
+    local -A checks
+    while read opt format expected ; do
+        opts+=("--$opt")
+        formats["--$opt"]=$format
+        checks["--$opt"]=$expected
+    done < <(parse_table "$tests")
+
+    # Now do the update in one fell swoop
+    run_podman update "${opts[@]}" $ctrname
+
+    # ...and check one by one
+    defer-assertion-failures
+    for opt in "${opts[@]}"; do
+        read expect <<<"${checks[$opt]}"
+        read format <<<"${formats[$opt]}"
+
+        run_podman inspect $ctrname --format $format
+        assert "$output" == "$expect" "$opt"
+    done
+    immediate-assertion-failures
+
+    # Clean up
+    run_podman rm -f -t0 $cid
+
+}
+
+# bats test_tags=ci:parallel
+@test "podman update - test HealtCheck flags without HealthCheck commands" {
+    local ctrname="c-h-$(safename)"
+
+    # flag-name=value
+    tests="
+    health-interval=10s
+    health-retries=5
+    health-timeout=10s
+    health-start-period=10s
+    health-startup-interval=10s
+    health-startup-retries=5
+    health-startup-success=10
+    health-startup-timeout=10s
+    "
+
+    run_podman run -d --name $ctrname $IMAGE top
+    cid="$output"
+
+    defer-assertion-failures
+    while read opt format expected ; do
+        run_podman 125 update "--$opt" $ctrname
+    done < <(parse_table "$tests")
```

Comment:
`$tests` is not a table, it's a simple list. No need for `parse_table` (nor the unused/confusing `format expected`). Better:
```bash
    for opt in $tests; do
        run_podman 125 update "--$opt" $ctrname
        assert "$output" =~ "healthcheck command is not set" "--$opt with no startup"
    done
```
Note that we never ever ever ever do just error-status checks without checking error messages. Please drill that habit into your head, it is one of the most important testing habits. Just checking "125" could just as easily fail with "unknown flag".